pandas/core/series.py

```diff
@@ -4667,7 +4671,9 @@ def rename(
                 errors=errors,
             )
         else:
-            return self._set_name(index, inplace=inplace)
+            return self._set_name(
+                index, inplace=inplace, deep=copy and not using_copy_on_write()
```

Comment:
Any reason we are doing this here and not in ``_set_name``?

Reply:
Simply passing copy through should be sufficient though

Reply:
It's to ignore copy=True, when Copy on Write is activated. I just copy pasted from your code in #51464

Reply:
Forgot about this, thx.

Reply:
Nothing else calls ``_set_name`` other than ``rename``, so I just left the Copy on Write logic in ``rename`` itself. I can move it if you want me to.

Reply:
Probably more consistent to push it down, but not a really strong opinion